core/platform/auth/firebase_auth_services.py

```
@@ -184,6 +185,124 @@ def get_all_external_accounts() -> List[auth_domain.ExternalAccount]:
+    errors = []
+    error_count = 0
+    batch_offset = 0
+
+    # NOTE: Convert None to '' to satisfy type checker (expects iter[str]).
+    # SDK raises ValueError for both None and '', so behavior is unchanged.
+    auth_ids = (account.auth_id or '' for account in accounts)
+
+    while batch := list(itertools.islice(auth_ids, feconf.FIREBASE_BATCH_SIZE)):
+        try:
+            result = firebase_auth.delete_users(batch)
+        except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)